apps/examples/media_tash/media_tash_main.c

```diff
@@ -0,0 +1,420 @@
+
+			ret = media_set_vol(vol);
+			if (ret != MEDIA_OK) {
+				printf("Error: media_set_vol\n");
+			}
+			else {
+				printf("Current Volume: %d\n", vol);
+			}
+		}
+		break;
+		case '7': {
+			pos += 1000;
+			if (pos > media_len - 1) {
+				pos = media_len - 1;
+			}
+
+			ret = media_set_position(pos);
+			if (ret != MEDIA_OK) {
+				printf("Error: media_set_position\n");
+			}
+		}
+		break;
+		case '8': {
+			pos -= 1000;
+			if (pos < 0) {
+				pos = 0;
+			}
+
+			ret = media_set_position(pos);
+			if (ret != MEDIA_OK) {
+				printf("Error: media_set_position\n");
+			}
+		}
+		break;
+		case 'a': {
+			if (media_is_playing()) {
+				media_stop_play();
+				disconnect_socket(&fd);
+			}
+
+			if (media_record_init() != RECORD_OK) {
+				printf("Error: media_record_init()\n");
+				return 0;
+			}
+			else {
+				printf("MediaRecord Init Okay..\n");
+			}
+ 
+			fd = open("/mnt/record", O_TRUNC | O_CREAT | O_RDWR);
+			if (fd < 0) {
+				printf("Cannot create file!\n");
+				DISCONNECT_WIFI_AND_CLOSE_SOCKET();
+				return 0;
+			} 
+
+			if (media_record_set_config(2, 16000, 0, MEDIA_FORMAT_PCM) != RECORD_OK) {
+				printf("Error: media_record_set_config()\n");
+				DISCONNECT_WIFI_AND_CLOSE_SOCKET();
+				return 0;
+			}
+
+			if (media_record_prepare() != RECORD_OK) {
+				printf("Error: media_record_prepare()\n");
+				DISCONNECT_WIFI_AND_CLOSE_SOCKET();
+				return 0;
+			}
+
+			if (media_record(fd) != RECORD_OK) {
+				printf("Error: media_record()\n");
+				DISCONNECT_WIFI_AND_CLOSE_SOCKET();
+				return 0;
+			}
+		}
+		break;
+		case 'b': {
+			if (media_is_playing()) {
+				media_stop_play();
+				disconnect_socket(&fd);
+			}
+
+			if (media_record_init() != RECORD_OK) {
+				printf("Error: media_record_init()\n");
+				return 0;
+			}
+			else {
+				printf("MediaRecord Init Okay..\n");
+			}
+
+			printf("Socket connecting...\n");
+			fd = connect_socket("192.168.1.213", 8080);
+			if (fd < 0) {
+				printf("Connection failed...\n");
+				dm_conn_wifi_disconnect();
+				return 0;
+			}
+			printf("Socket connected successfully.\n");
+
+			if (media_record_set_config(2, 16000, 0, MEDIA_FORMAT_PCM) != RECORD_OK) {
+				printf("Error: media_record_set_config()\n");
+				DISCONNECT_WIFI_AND_CLOSE_SOCKET();
+				return 0;
+			}
+
+			if (media_record_prepare() != RECORD_OK) {
+				printf("Error: media_record_prepare()\n");
+				DISCONNECT_WIFI_AND_CLOSE_SOCKET();
+				return 0;
+			}
+
+			if (media_record(fd) != RECORD_OK) {
+				printf("Error: media_record()\n");
+				DISCONNECT_WIFI_AND_CLOSE_SOCKET();
+				return 0;
+			}
+		}
+		break;
+		case 'c': {
+			if (media_pause_record() != RECORD_OK) {
+				printf("Error: media_pause_record()\n");
+			}
+		}
+		break;
+		case 'd': {
+			if (media_resume_record() != RECORD_OK) {
+				printf("Error: media_resume_record()\n");
+			}
+		}
+		break;
+		case 'e': {
+			if (media_stop_record() != RECORD_OK) {
+				printf("Error: media_stop_record()\n");
+			}
+		}
+		break;
+		case 'f': {
```

Comment:
why don't you use media_play instead of using tinyalsa directly?

Reply:
Yes, I agree, we should avoid to call tinyalsa api directly.